docs/src/reference/sparsity_pattern.md

```diff
@@ -0,0 +1,80 @@
+# Sparsity pattern and sparse matrices
+
+This is the reference documentation for sparsity patterns and sparse matrix instantiation.
+
+A general overview of the usage is as follows:
+ 1. Create an empty pattern using a constructor, for example [`SparsityPattern`](@ref
+    SparsityPattern(::Int, ::Int))
+ 2. Add entries to the pattern from the DofHandler using [`create_sparsity_pattern!`](@ref)
+ 3. Condense the pattern using the ConstraintHandler using
+    [`condense_sparsity_pattern!`](@ref)
+ 4. Instantiate sparse matrices from the sparsity pattern using [`create_matrix`](@ref)
+
+For example, the steps above could look as follows when creating a `SparseMatrixCSC` matrix:
+```julia
+dh = DofHandler(...)
+ch = ConstraintHandler(...)
+# 1. Create empty pattern
+sparsity_pattern = SparsityPattern(ndofs(dh), ndofs(dh))
+# 2. Add entries
+create_sparsity_pattern!(sparsity_pattern, dh)
```

Comment:
Maybe we should mention around here how to manually add entries (via `Ferrite.add_entry!`).